APM/CommandSubscriberRegistry.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Bundle\MongoDBBundle\APM;
+
+use MongoDB\Driver\Monitoring\CommandSubscriber;
+use function method_exists;
+
+final class CommandSubscriberRegistry
+{
+    /** @var CommandSubscriber[] */
+    private $commandSubscribers;
+
+    public function __construct(iterable $commandSubscribers)
+    {
+        $this->commandSubscribers = $commandSubscribers;
```

Comment:
Would it make sense for this to validate that each element within `$commandSubscribers` is actually a CommandSubscriber instance?

Reply:
On second thought, I wonder if this is instead supposed to accept only [`Doctrine\ODM\MongoDB\APM\CommandLogger`](https://github.com/doctrine/mongodb-odm/blob/6ded0705dba3661438b501bfaa39f42858293137/lib/Doctrine/ODM/MongoDB/APM/CommandLogger.php) instances, as that class is where `register()` and `unregister()` methods are defined.

Reply:
We'll have to introduce a new interface, as the `CommandLogger` class is `final` but we also need a command logger that logs to any PSR-compatible logger. I'd introduce a `CommandLogger` interface that extends `CommandSubscriber` and exposes `register` and `unregister` methods, then limit possible arguments to this method to that interface.